example/main.go

```diff
@@ -29,7 +29,9 @@ type DatabaseConfig struct {
 
 type Config struct {
 	Debug    bool           `env:"DEBUG"`
+	Host     string         `env:"HOST,default=localhost"`
 	Port     string         `env:"PORT,default=8080"`
+	Address  string         `env:"ADDRESS,default=${HOST}:$PORT,expand"`
```

Comment:
You should use different name here than host and port, because the example might let people think HOST and PORT already need to be provided as fields

So maybe, something like


```suggestion
	Address  string         `env:"ADDRESS,default=${ADDR}:$BIND,expand"`
```

Reply:
This is pretty standard variable expansion behavior, take Bash for example:

```sh
HOST="localhost"
PORT="8080"
ADDR="${HOST}:${PORT}"
echo $ADDR
```

In bash, you _do_ define them in order. Since variables are expected to be defined before you start your code, IE:

```
HOST=localhost PORT=8080 go run main.go
```

The behavior of `expand` would work with that.

The reverse lookup of default values also allows for the code to be able to leverage those, so if values aren't set in the environment, you don't have to run into errors.

Reply:
good point about the default value for variable used in another variable using expand.

It should be covered by a dedicated test maybe

Setting just

PORT="8081"

and expect to have

```go
type Config struct {
	Debug    bool           `env:"DEBUG"`
	Host     string         `env:"HOST,default=localhost"`
	Port     string         `env:"PORT,default=8080"`
	Address  string         `env:"ADDRESS,default=${HOST}:$PORT,expand"`
}
```

```
cfg := Config{
	Host: "localhost"
	Port: "8081"
	Address: "localhost:8081"
}
```